services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowDiversitySpec.groovy

```diff
@@ -0,0 +1,326 @@
+        def flow3SwitchCount = pathHelper.getInvolvedSwitches(flow3.id).size()
+        def expectedValuesMap = [
+                diverseGroup: [
+                        (flow1.id): [islCount: 0, switchCount: 2, islPercent: 0, switchPercent: 100],
+                        (flow2.id): [islCount     : 0, switchCount: 2, islPercent: 0,
+                                     switchPercent: (2 * 100 / flow2SwitchCount).toInteger()],
+                        (flow3.id): [islCount     : 0, switchCount: 2, islPercent: 0,
+                                     switchPercent: (2 * 100 / flow3SwitchCount).toInteger()]
+                ],
+                otherFlows  : [
+                        (flow1.id): [
+                                (flow2.id): [islCount: 0, switchCount: 2, islPercent: 0, switchPercent: 100],
+                                (flow3.id): [islCount: 0, switchCount: 2, islPercent: 0, switchPercent: 100]
+                        ],
+                        (flow2.id): [
+                                (flow1.id): [islCount     : 0, switchCount: 2, islPercent: 0,
+                                             switchPercent: (2 * 100 / flow2SwitchCount).toInteger()],
+                                (flow3.id): [islCount     : 0, switchCount: 2, islPercent: 0,
+                                             switchPercent: (2 * 100 / flow2SwitchCount).toInteger()]
+                        ],
+                        (flow3.id): [
+                                (flow1.id): [islCount     : 0, switchCount: 2, islPercent: 0,
+                                             switchPercent: (2 * 100 / flow3SwitchCount).toInteger()],
+                                (flow2.id): [islCount     : 0, switchCount: 2, islPercent: 0,
+                                             switchPercent: (2 * 100 / flow3SwitchCount).toInteger()]
+                        ]
+                ]
+        ]
+        [flow1Path, flow2Path, flow3Path].each { flowPath ->
+            with(flowPath.diverseGroupPayload) { diverseGroup ->
+                with(diverseGroup.overlappingSegments) {
+                    verifyAll {
+                        islCount == expectedValuesMap["diverseGroup"][flowPath.id]["islCount"]
+                        switchCount == expectedValuesMap["diverseGroup"][flowPath.id]["switchCount"]
+                        islPercent == expectedValuesMap["diverseGroup"][flowPath.id]["islPercent"]
+                        switchPercent == expectedValuesMap["diverseGroup"][flowPath.id]["switchPercent"]
+                    }
+                }
+                with(diverseGroup.otherFlows) { otherFlows ->
+                    otherFlows.each { flow ->
+                        assert flow.id in ([flow1, flow2, flow3]*.id - flowPath.id)
+                        with(flow.segmentsStats) {
+                            verifyAll {
+                                islCount == expectedValuesMap["otherFlows"][flowPath.id][flow.id]["islCount"]
```

Comment:
`[flowPath.id][flow.id]`
how flow path id is different from flow id? I don't understand this. I see that these are params from different loops, but I think the names of these vars are strange

Reply:
Ok, got it. I think renaming `flow.id` to `otherFlow.id` will make it more clear, since `flow.id` unintentionally associates with the main `flowid` in that json